src/lem/mod.rs

```diff
@@ -108,22 +109,22 @@ impl MetaPtr {
 }
 
 /// The basic building blocks of LEMs.
-#[derive(Clone)]
+#[derive(Clone, PartialEq)]
 pub enum LEMOP {
```

Comment:
Just FYI, since this is changing, are you aware of the `#[non_exhaustive]` attribute?
https://doc.rust-lang.org/reference/attributes/type_system.html